### PR TITLE
Zdx auth with token

### PR DESCRIPTION
**Pull request by Rlhirmes103**

### CODE REVIEW

File: zendesk/zendesk.go [hunk 1]
```diff
@@ -134,6 +137,33 @@ func NewURLClient(endpoint, username, password string, middleware ...MiddlewareF
 	return c, nil
 }
 
+// New acceps an access token and domain to create a new Client
+func New(token *oauth2.Token, base string) (Client, error) {
+
+	var err error
+	if token == nil {
+		err = fmt.Errorf("No auth token")
```
Comment (sdalezman) [inline zendesk/zendesk.go]: return the error directly

``` golang
if token == nil {
     return nil, errors.New("auth token is nil")
}
```
also if you don't have any error formatting it's better practice to return errors.New directly

File: zendesk/zendesk.go [hunk 2]
```diff
@@ -134,6 +137,33 @@ func NewURLClient(endpoint, username, password string, middleware ...MiddlewareF
...
+
+	domain := fmt.Sprintf("https://%s.zendesk.com", base)
+	baseURL, err := url.Parse(domain)
+	if err != nil {
+		return nil, err
+	}
+
+	tok := token.AccessToken
+	s := &client{
+		token:     tok,
```
Comment (sdalezman) [inline zendesk/zendesk.go]: you can set `token: token.AccessToken`

File: zendesk/zendesk.go [hunk 3]
```diff
@@ -161,8 +191,11 @@ func (c *client) request(method, endpoint string, headers map[string]string, bod
 		return nil, err
 	}
 
+	bearer := fmt.Sprintf("Bearer %s", c.token)
```
Comment (sdalezman) [inline zendesk/zendesk.go]: what if the user is using the basic auth approach?

Comment (Rlhirmes103) [reply zendesk/zendesk.go]: basic auth should still work here

File: zendesk/zendesk.go [hunk 4]
```diff
@@ -134,6 +137,30 @@ func NewURLClient(endpoint, username, password string, middleware ...MiddlewareF
 	return c, nil
 }
 
+// New acceps an access token and domain to create a new Client
+func New(token *oauth2.Token, base string) (Client, error) {
```
Comment (sdalezman) [inline zendesk/zendesk.go]: do you need to utilize the refresh token here?

Comment (Rlhirmes103) [reply zendesk/zendesk.go]: i tested it like this and it worked...

File: zendesk/zendesk.go (hunk 3 above)
Comment (Rlhirmes103) [reply zendesk/zendesk.go]: @sdalezman  resolved this to work with basic auth

**Comment by Rlhirmes103** [comment]
@sdalezman  i dont have a circleci account to see why the branches are failing

**Comment by Rlhirmes103** [comment]
@sdalezman  finally!!

File: zendesk/job_status.go [hunk 5]
```diff
@@ -6,13 +6,13 @@ import "fmt"
...
-	Message  *string  `json:"message,omitempty"`
-	Progress *int64   `json:"progress,omitempty"`
-	Results  []Result `json:"results,omitempty"`
-	Status   *string  `json:"status,omitempty"`
-	Total    *int64   `json:"total,omitempty"`
-	URL      *string  `json:"url,omitempty"`
+	ID       *string `json:"id,omitempty"`
+	Message  *string `json:"message,omitempty"`
+	Progress *int64  `json:"progress,omitempty"`
+	Results  Result  `json:"results,omitempty"`
```
[the remark this answers is not in the document]
Comment (Rlhirmes103) [reply zendesk/job_status.go]: this was here from the sdk but can change it

**Comment by sdalezman** [comment]
@Rlhirmes103 tests are failing here

**Pull request merged**